src/term/mod.rs

```diff
@@ -200,6 +200,65 @@ pub use self::mode::TermMode;
 
 pub const TAB_SPACES: usize = 8;
 
+#[derive(Clone, Copy)]
+enum Charset {
+    Ascii,
+    /// A sparse charset only has a few characters in it and is best mapped with
+    /// a match statement
+    Sparse(StandardCharset),
+    /// A dense charset has most of the 96 possible characters in it and may be
+    /// better handler by a lookup table
+    Dense(StandardCharset),
+    /// A user defined charset handled by some more complicated mechanism
+    User,
+}
+
+impl Charset {
+    /// Switch/Map character to the active charset. Ascii is the common case and
+    /// for that we want to do as little as possible.
+    #[inline]
+    fn switch(&self, c: char) -> char {
+        match *self {
+            Charset::Ascii => c,
+            Charset::Sparse(set) => match set {
+                StandardCharset::LineDrawing =>
+                    match c {
```

Comment:
I am not entirely sure what you mean, but from looking at the source for a few terminal emulators I found links to http://vt100.net/docs/vt102-ug/table5-13.html and http://www.mirrorservice.org/sites/www.bitsavers.org/pdf/dec/terminal/vt340/EK-VT3XX-TP-001_VT330_VT340_Text_Programming_Mar87.pdf (page 41 for technical character set). 

In terms of the implementation I was debating whether to use an array and index into it using the first byte of _c_ with an offset, or just use _c_ in a match statement, the latter definitely makes more sense if only a few characters are different from ASCII (sparse mapping), but maybe not if all the chars are different (dense mapping) or it needs to be configurable.